sputnikdao-factory2/src/lib.rs

```diff
@@ -126,15 +126,15 @@ impl SputnikDAOFactory {
         }
     }
 
-    /// Tries to upgrade given account created by this factory to the default code.
-    pub fn upgrade(&self, account_id: AccountId) {
+    /// Tries to update given account created by this factory to the specified code.
+    pub fn update(&self, account_id: AccountId, code_hash: Base58CryptoHash) {
```

Comment:
Why the method name change here? I believe this to cause more issues than needed, esp for clients.

Reply:
Because no one can stop you from downgrading a DAO to a previous version. If you provide the right hash, you can also downgrade.

Reply:
I get that, but changing the method name is not good, is it absolutely necessary? I dont want clients supporting 2 method names for the same thing.

Reply:
Why do you need to support 2 method names? `upgrade` method was added very recently, it's not part of the deployed factory yet.

This is how the flow will go:
`dao::UpgradeSelf { hash } -> dao::upgrade_using_factory -> factory::update() -> dao::update()`

DAOs only need to know about `UpgradeSelf { hash }`, nothing more than that.

Reply:
Gotcha, didn't know this was a new method as well. In any case - update or upgrade is fine, i just want to be absolutely clear that downgrading is not supported, which can be implied with good naming.